include/JSystem/JKernel/JKRAssertHeap/JKRAssertHeap.h

```diff
@@ -10,23 +10,23 @@ class JKRAssertHeap : public JKRHeap {
     virtual ~JKRAssertHeap();
 
 public:
-    /* vt[04] */ virtual u32 getHeapType(void);           /* override */
-    /* vt[05] */ virtual bool check(void);                /* override */
-    /* vt[06] */ virtual bool dump_sort(void);            /* override */
-    /* vt[07] */ virtual bool dump(void);                 /* override */
-    /* vt[08] */ virtual void do_destroy(void);           /* override */
-    /* vt[09] */ virtual void* do_alloc(u32, int);        /* override */
-    /* vt[10] */ virtual void do_free(void*);             /* override */
-    /* vt[11] */ virtual void do_freeAll(void);           /* override */
-    /* vt[12] */ virtual void do_freeTail(void);          /* override */
-    /* vt[13] */ virtual void do_fillFreeArea(void);      /* override */
-    /* vt[14] */ virtual s32 do_resize(void*, u32);       /* override */
-    /* vt[15] */ virtual s32 do_getSize(void*);           /* override */
-    /* vt[16] */ virtual s32 do_getFreeSize(void);        /* override */
-    /* vt[17] */ virtual void* do_getMaxFreeBlock(void);  /* override */
-    /* vt[18] */ virtual s32 do_getTotalFreeSize(void);   /* override */
-    /* vt[19] */ virtual u8 do_changeGroupID(u8 param_1); /* override */
-    /* vt[20] */ virtual u8 do_getCurrentGroupId(void);   /* override */
+    /* vt[04] */ virtual u32 getHeapType(void) const;          /* override */
+    /* vt[05] */ virtual bool check(void);                     /* override */
+    /* vt[06] */ virtual bool dump_sort(void);                 /* override */
+    /* vt[07] */ virtual bool dump(void);                      /* override */
+    /* vt[08] */ virtual void do_destroy(void);                /* override */
+    /* vt[09] */ virtual void* do_alloc(u32, int);             /* override */
+    /* vt[10] */ virtual void do_free(void*);                  /* override */
```

Comment:
```suggestion
    /* vt[10] */ void do_free(void*);                  /* override */
```
Do these need to be marked as virtual?

Reply:
For function that are overridden, virtual is not required. With both the vt[XX] and override comment maybe the virtual keyword is not needed, but this is how I done it for everything else in JKernel. Best would be to use the override keyword, but that was release in C++11, which MWCC ofcourse doesn't support. I am fine with doing it either way as this doesn't really effect the matching of functions. But maybe it's better to change it later, in all classes at once, when we have decided on the style we want.